routes/mentor.go

```diff
@@ -3,8 +3,45 @@ package routes
 import (
 	"fmt"
 	"net/http"
+	"encoding/json"
+	"io/ioutil"
+	"github.com/jinzhu/gorm"
+	"kwoc20-backend/models"
 )
 
 func MentorReg(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "hello")
+
+	var mentor models.Mentor
+	body, _ := ioutil.ReadAll(r.Body)
+	err := json.Unmarshal(body, &mentor)
+	if err != nil {
+	  http.Error(w, err.Error(), 400)
+	  return 		
+	}
+
+	
+	db, err := gorm.Open("sqlite3", "kwoc.db")
+	if err != nil {
+		 http.Error(w, err.Error(), 500)
```

Comment:
Style clash with Line 43.
If you are using `w.WriteHeader` why bother using `http.Error`? (I maybe wrong here)

Use something like `w.WriteHeader(http.StatusInternalServerError)`

Reply:
https://golang.org/doc/articles/wiki/#tmp_9
http.Error was used in case of golang docs

hence using it seemed better

ya u are rite, uniformity is also important!
what do u say?

Reply:
Use a single style. Either `http.Error` or `w.WriteHeader`. `http.Error` sends the error message. But since we shall be sending json, sending raw messages such as that won't be conforming to our Content-type header. Also I'd rather not use `500` directly, but `http.StatusInternalServerError` (docs also have them).
So either use:
```go
w.WriteHeader(http.StatusInternalServerError)
w.Write([]bytes(`{"message": "` + err.Error() + `"}`)
return
```

or

```go
http.Error(w, `{"message": "` + err.Error() + `"}`, http.StatusInternalServerError)
return
```

Take your pick.

Reply:
```
w.WriteHeader(http.StatusInternalServerError)
w.Write([]bytes(`{"message": "` + err.Error() + `"}`)
return
```
contains an extra line, but this this is good for sake of uniformity
lets fix this !